src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavacParser.java

```diff
@@ -3723,6 +3723,10 @@ JCVariableDecl variableDeclaratorId(JCModifiers mods, JCExpression type, boolean
                 JCExpression pn = qualident(false);
                 if (pn.hasTag(Tag.IDENT) && ((JCIdent)pn).name != names._this) {
                     name = ((JCIdent)pn).name;
+                } else if (lambdaParameter) {
```

Comment:
I wonder if we could help the error recovery a little bit more, like e.g.:
```
                } else if (lambdaParameter && type == null) {
                    // we have a lambda parameter that is not an identifier this is a syntax error
                    type = pn;
                    name = names.empty;
                    reportSyntaxError(pos, Errors.Expected(IDENTIFIER));
                } else {
```

Which would keep the parsed thing as a type for the lambda parameter?

Reply:
yep I like this suggestion. I have updated the PR accordingly, thanks!